src/test/java/org/elasticsearch/test/rest/ElasticsearchRestTestCase.java

```diff
@@ -158,23 +164,27 @@ protected Settings nodeSettings(int nodeOrdinal) {
     }
 
     private static List<RestTestCandidate> collectTestCandidates(int id, int count) throws RestTestParseException, IOException {
-        String[] paths = resolvePathsProperty(REST_TESTS_SUITE, DEFAULT_TESTS_PATH);
-        Map<String, Set<Path>> yamlSuites = FileUtils.findYamlSuites(DEFAULT_TESTS_PATH, paths);
-
         List<RestTestCandidate> testCandidates = Lists.newArrayList();
-        RestTestSuiteParser restTestSuiteParser = new RestTestSuiteParser();
-        //yaml suites are grouped by directory (effectively by api)
-        for (String api : yamlSuites.keySet()) {
-            List<Path> yamlFiles = Lists.newArrayList(yamlSuites.get(api));
-            for (Path yamlFile : yamlFiles) {
-                String key = api + yamlFile.getFileName().toString();
-                if (mustExecute(key, id, count)) {
-                    RestTestSuite restTestSuite = restTestSuiteParser.parse(api, yamlFile);
-                    for (TestSection testSection : restTestSuite.getTestSections()) {
-                        testCandidates.add(new RestTestCandidate(restTestSuite, testSection));
+        FileSystem fileSystem = getFileSystem();
+        try {
```

Comment:
put some comment on it that it can be null or so I can already see somebody changing it and tests won't catch it.